hyp3lib/get_orb.py

```diff
@@ -1,197 +1,147 @@
-
-
-def downloadSentinelOrbitFileProvider(granule, provider, directory):
-  if provider.upper() == 'ASF':
-    urlOrb, fileNameOrb = getOrbFile(granule)
-    verify = True
-  elif provider.upper() == 'ESA':
-    urlOrb, fileNameOrb = getOrbitFileESA(granule)
-    verify = False
-
-  if directory:
-    stateVecFile = os.path.join(directory, fileNameOrb)
-  else:
-    stateVecFile = fileNameOrb
-
-  if not os.path.isfile(stateVecFile):
-    if len(stateVecFile) > 0:
-      fetchOrbitFile(urlOrb,stateVecFile,verify)
-      return stateVecFile
-    else:
-      return None
-  else:
-    print("Using existing orbit file; provider unknown")
-    return stateVecFile
-
-
-def downloadSentinelOrbitFile(granule,directory=None):
+    best = None
+    for file in file_list:
+        file = file.strip()
+        t = re.split('_', file.replace('T', '').replace('V', ''))
+        if len(t) > 7:
+            start = t[6]
+            end = t[7].replace('.EOF', '')
+            if start < timestamp < end:
+                d = ((int(timestamp) - int(start)) + (int(end) - int(timestamp))) / 2
+                if d > d1:
+                    best = file
+                    d1 = d
+    return search_url + best
+
+
+def get_orbit_url(granule: str, orbit_type: str = 'AUX_POEORB', provider: str = 'ESA'):
+    """Get the URL of an orbit file from a provider
+
+    Args:
+        granule: Granule name to find an orbit file for
+        orbit_type: Orbit type to download
+        provider: Povider name to download the orbit file from
+
+    Returns:
+        orbit_url: The url to the matched orbit file
+    """
+    platform = granule[0:3]
+    time_stamps = re.split('_+', granule)[4:6]
+
+    if provider.upper() == 'ESA':
+        delta = timedelta(seconds=60)
+        start_time = datetime.strptime(time_stamps[0], '%Y%m%dT%H%M%S') - delta
+        end_time = datetime.strptime(time_stamps[1], '%Y%m%dT%H%M%S') + delta
+
+        params = {
+            "product_type": orbit_type.upper(),
+            "product_name__startswith": platform,
+            "validity_start__lt": start_time.strftime('%Y-%m-%dT%H:%M:%S'),
+            "validity_stop__gt": end_time.strftime('%Y-%m-%dT%H:%M:%S'),
+            "ordering": "-creation_date",
+            "page_size": "1",
+        }
+
+        response = requests.get(url='https://qc.sentinel1.eo.esa.int/api/v1/', params=params)
+        response.raise_for_status()
+        qc_data = response.json()
+
+        orbit_url = None
+        if qc_data["results"]:
+            orbit_url = qc_data["results"][0]["remote_url"]
+        return orbit_url
+
+    elif provider.upper() == 'ASF':
+        orbit_url = _get_asf_orbit_url(orbit_type.lower(), platform, time_stamps[0].replace('T', ''))
+        return orbit_url
+
+    raise OrbitDownloadError(f'Unknown orbit file provider {provider}')
+
+
+def _download_and_verify_orbit(url: str, directory: str = ''):
+    orbit_file = download_file(url, directory=directory, chunk_size=5242880)
```

Comment:
should drop `chunk_size=5242880` here, don't need a multi-part download for text files